### PR TITLE
Commit of steps 0-3 in Music App

### PR DESCRIPTION
**Pull request by falese**
I've tried everything I can think of to get the partials and my routes working with songs, I'm still getting either a nil class error, or a route doesn't exist error.  I did get the models wired up and everything styled in bootstrap though.

UPDATED

I got all models working correctly.

### CODE REVIEW

**Comment by tgaff** [comment]
awesome use of commits for each step!

**Comment by falese** [comment]
Thanks! its a good way to remember what i did!

**Comment by tgaff** [comment]
Hi @falese there are 500+ files in here.  It looks like you may have accidentally added all of the tmp/* and log/* files.  Would you mind removing those?  Make sure in the future you do `git add .` or `git add "name of file"`.  

I'll keep reviewing it with those in place but I don't want to merge them in; so let me know when you get rid of them.

**Comment by tgaff** [comment]
Actually it looks like on the "files" tab github won't show me the files I need to comment on (it says there's too many). :-/

**Comment by falese** [comment]
Hmm OK. Not sure how that happened. I'll clean up

**Comment by tgaff** [comment]
Thank you!

**Comment by falese** [comment]
Ok lets try this again.

It might be a cool topic to go over how to use the gitignore file

**Comment by tgaff** [comment]
That looks a lot better! thanks!

Rails comes with a .gitignore that works.  However if you add the tmp files before the .gitignore they'll already be there and it takes another commit to remove them. I think maybe when you moved your app to this repo you moved those files too and committed them all.  :-/

File: SeanFalese-falese/music_app/my_music_app/app/views/layouts/_main_nav.html.erb [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<nav class="navbar navbar-custom navbar-fixed-top" role="navigation">
+    <div class="container">
+        <div class="navbar-header">
+            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-main-collapse">
+                <i class="fa fa-bars"></i>
+            </button>
+            </a>
+        </div>
+
+        <!-- Collect the nav links, forms, and other content for toggling -->
```
Comment (tgaff) [inline SeanFalese-falese/music_app/my_music_app/app/views/layouts/_main_nav.html.erb]: Your navbar looks good on each page.  You might want to remove the about link until it has a page to go to

File: SeanFalese-falese/music_app/my_music_app/app/controllers/reviews_controller.rb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+  def destroy
+  end
+
+end
+
+private
+def set_reviewable_item
+  reviewable_item = [Artist, Song].detect{|c| params["#{c.name.underscore}_id"]}   #this figures out which class the request comes from
+        @reviewable_item = reviewable_item.find(params["#{reviewable_item.name.underscore}_id"]) #this sets the appropriate foreign key
```
Comment (tgaff) [inline SeanFalese-falese/music_app/my_music_app/app/controllers/reviews_controller.rb]: nice!  Does everything here make sense?  

* for long comments you can just put them on the line before the code
* also there are a few other ways to do polymorphics so if you're researching it keep in mind that the guides may be different dependent on the author's preferred approach.
* There's a good railscast for it too, but it's done on rails3.  You'll probably be able to spot the differences pretty quickly, but just FYI.

File: SeanFalese-falese/music_app/my_music_app/config/routes.rb [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  # The priority is based upon order of creation: first created -> highest priority.
+  # See how all your routes lay out with "rake routes".
+
+  # You can have the root of your site routed with "root and thats what i did below"
+  root 'static_pages#home'
+  resources :artists do
+    resources :songs
+    resources :reviews
+  end
+  resources :songs do
```
Comment (tgaff) [inline SeanFalese-falese/music_app/my_music_app/config/routes.rb]: you may not want this (or maybe it's something you're working on).  When I visit /songs it looks unfinished.

Comment (falese) [inline SeanFalese-falese/music_app/my_music_app/config/routes.rb]: Yup I'm still working on that. The plan is to list all songs and be able to
add a song and select from a drop down contain a collection of artists.

File: SeanFalese-falese/music_app/my_music_app/app/views/artists/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+</p>
+<div class ="table-responsive">
+  <table class="table table-hover">
+
+    <tr>
+      <th>Title</th>
+      <th>Length</th>
+      <th>Genre</th>
+    </tr>
+    <% @artist.songs.each do |song| %>
```
Comment (tgaff) [inline SeanFalese-falese/music_app/my_music_app/app/views/artists/show.html.erb]: An alternative might be to make this into a songs partial; then later you can use it on your songs index as well.

**Comment by tgaff** [comment]
I didn't spot anyway to edit or delete a song?  I'm guessing you're still working on that and reviews though.  Let us know if you need help.  Overall it's coming along quite nicely.  Your bootstrap looks pretty good, although you might want to get rid of the highlight when you mouse over (maybe I'm picky).
Great job overall!

**Comment by falese** [comment]
Thanks!  I am working on the edit and deletes for songs and reviews. My
plan is to make the table row clickable (jquery) to show the record and
have edit buttons on the show form.

**Pull request merged**